packages/core/schematics/migrations/ngclass-to-class-migration/util.ts

```diff
@@ -0,0 +1,497 @@
+    const fullText = sourceFile.getFullText();
+
+    // Get the position of the 'imports' property
+    const propertyStart = importsProperty.getStart();
+    const propertyEnd = importsProperty.getEnd();
+
+    // Find the start of the line containing 'imports'
+    const textBefore = fullText.substring(0, propertyStart);
+    const lineStart = textBefore.lastIndexOf('\n') + 1;
+
+    // Find the end of the line or next property
+    let lineEnd = fullText.indexOf('\n', propertyEnd);
+    if (lineEnd === -1) {
+      lineEnd = fullText.length;
+    }
+
+    // Check if there's a trailing comma after the property
+    const textAfter = fullText.substring(propertyEnd, lineEnd);
+    const commaMatch = textAfter.match(/^\s*,/);
+
+    if (commaMatch) {
+      // Include the comma in the removal
+      end = propertyEnd + commaMatch[0].length;
+    } else {
+      // Check if there's a leading comma before the property
+      const beforeText = fullText.substring(lineStart, propertyStart);
+      const leadingCommaMatch = beforeText.match(/,\s*$/);
+      if (leadingCommaMatch) {
+        start = propertyStart - leadingCommaMatch[0].length;
+      } else {
+        end = propertyEnd;
+      }
+    }
+
+    // Remove the entire line if it only contains the imports property
+    const lineText = fullText.substring(lineStart, lineEnd).trim();
+    if (lineText.startsWith('imports:') || lineText.includes('imports: [')) {
+      start = lineStart;
+      end = lineEnd;
+    } else {
+      start = propertyStart;
+    }
+  }
+  // Handle commas to avoid leaving trailing commas or syntax errors.
+  else if (elements.length > 1) {
+    if (ngClassIndex === 0) {
+      // First element: remove up to the start of the next element (including comma).
+      const nextElement = elements[ngClassIndex + 1];
+      const sourceFile = ngClassElement.getSourceFile();
+      const text = sourceFile.text.substring(end, nextElement.getStart());
+      const commaMatch = text.match(/^\s*,/);
+      if (commaMatch) {
+        end = end + commaMatch[0].length;
+      }
+    } else {
+      // Middle or last element: remove from the end of the previous element (including comma).
+      const prevElement = elements[ngClassIndex - 1];
+      const sourceFile = ngClassElement.getSourceFile();
+      const text = sourceFile.text.substring(prevElement.getEnd(), start);
+      const commaMatch = text.match(/,\s*$/);
+      if (commaMatch) {
+        start = prevElement.getEnd() + text.lastIndexOf(',');
+      }
+    }
+  }
+
+  return new Replacement(file, new TextUpdate({position: start, end, toInsert: ''}));
+}
+
+export function calculateImportReplacements(info: ProgramInfo, sourceFiles: Set<ts.SourceFile>) {
+  const importReplacements: Record<
+    ProjectFileID,
+    {add: Replacement[]; addAndRemove: Replacement[]}
+  > = {};
+
+  for (const sf of sourceFiles) {
+    const file = projectFile(sf, info);
+    const importManager = new ImportManager();
```

Comment:
The `importManager` can be shared accross files 

```suggestion
  const importManager = new ImportManager();
  for (const sf of sourceFiles) {
    const file = projectFile(sf, info);
```